### PR TITLE
General Refactor

### PR DESCRIPTION
**Pull request by RichMatthews**
I am aware this is a big PR. lots of files changed etc. the project is a lot neater now though. Still work to be done but have:
- removed unnecessary redux, component state
- have tidied up most of duplication
- have made styling nicer
- have extracted most things into constants and not strings
- have made it easier for PO's to add stuff to app

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -4,7 +4,9 @@
   "private": true,
   "dependencies": {
     "@material-ui/core": "^3.6.2",
+    "eslint-config-airbnb-base": "^13.1.0",
     "humps": "^2.0.1",
+    "prettier": "^1.16.4",
```
Comment (victor-paddle) [inline package.json]: do we need `prettier` and `eslint` as a production dependency?

Comment (RichMatthews) [reply package.json]: @victor-paddle do not review 🙈

Comment (victor-paddle) [reply package.json]: sorry

Comment (RichMatthews) [reply package.json]: have moved to dev dependencies now 👍

File: src/components/Modal/buttons/index.js [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+    z-index: 100;
+    height: 90%;
+`
+
+const Button = styled.button`
+    font-family: Montserrat;
+    cursor: pointer;
+    border-radius: 7px;
+    padding: 15px;
+    line-height: 1px;
```
Comment (rob-hunt-paddle) [inline src/components/Modal/buttons/index.js]: Why such a small line height?

File: src/components/dropdown/index.js [hunk 3]
```diff
@@ -1,68 +1,100 @@
...
-    componentDidMount() {
-      this.updateUrlParams(this.props.url.path)
-    }
+const RightHandSide = styled.div`
+    display: flex;
+    align-items: center;
+`
+
+export const Select = styled.select`
+    box-shadow: 0px 2px 2px rgba(0, 0, 0, 0.25);
```
Comment (rob-hunt-paddle) [inline src/components/dropdown/index.js]:
```suggestion
    box-shadow: 0 2px 2px rgba(0, 0, 0, 0.25);
```

File: src/components/dropdown/index.js [hunk 4]
```diff
@@ -1,68 +1,100 @@
...
+                <RightHandSide>
+                    <Select
+                        greyOutContainer={this.state.greyOutContainer}
+                        onChange={e => updateValue(labelName, e.target.value)}
+                    >
+                        {options.map(option => (
+                            <option
+                                selected={
+                                    option.value ===
+                                    codeToRender.contents.filter(parameter => parameter.name === labelName)[0].value
```
Comment (rob-hunt-paddle) [inline src/components/dropdown/index.js]: Worth using lodash get?

File: src/components/input/index.js [hunk 5]
```diff
@@ -1,53 +1,84 @@
...
+                        checked={this.props.codeToRender.contents.filter(param => param.name === labelName)[0].render}
+                    />
+                    <StyledLabel>{labelName} </StyledLabel>
+                </LeftHandSide>
+                <RightHandSide>
+                    <StyledInput
+                        greyOutContainer={this.state.greyOutContainer}
+                        type="text"
+                        onChange={e => updateValue(labelName, e.target.value)}
+                        value={codeToRender.contents.filter(par => par.name === labelName)[0].value}
```
Comment (rob-hunt-paddle) [inline src/components/input/index.js]: Worth using lodash get?

File: src/components/Modal/buttons/index.js [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+    display: flex;
+    flex-direction: column;
+    justify-content: space-between;
+    padding: 10px;
+    z-index: 100;
+    height: 90%;
+`
+
+const Button = styled.button`
+    font-family: Montserrat;
```
Comment (rob-hunt-paddle) [inline src/components/Modal/buttons/index.js]: Do we want to define a fallback?

File: src/components/Modal/buttons/index.js [hunk 7]
```diff
@@ -0,0 +1,72 @@
...
+    background: #ffb61b;
+`
+
+const CancelBtn = styled(Button)`
+    background: red;
+    border: none;
+`
+
+const Heading = styled.h3`
+    font-family: Montserrat;
```
Comment (rob-hunt-paddle) [inline src/components/Modal/buttons/index.js]: Do we want to define a fallback?

File: src/components/group/index.js [hunk 8]
```diff
@@ -1,11 +1,26 @@
...
+    }
+`
 
-export default group
+const GroupHeading = styled.h6`
+    margin: 0;
+    color: #0970b9;
+    font-size: 16px;
+    margin-bottom: 8px;
+    font-family: Montserrat;
```
Comment (rob-hunt-paddle) [inline src/components/group/index.js]: Do we want to define a fallback?

File: src/components/shared/index.js [hunk 9]
```diff
@@ -0,0 +1,5 @@
+import styled from 'styled-components'
+
+export const StyledLabel = styled.label`
+    font-family: Montserrat;
```
Comment (rob-hunt-paddle) [inline src/components/shared/index.js]: Do we want to define a fallback?

File: src/components/info/index.js [hunk 10]
```diff
@@ -1,14 +1,24 @@
...
-const info = ({labelName, tooltipText}) => (
-  <div data-tip="tooltip" data-type="info" data-place="left" data-for={labelName.replace(/\s/g, '')+"Tooltip"} className="info-icon">
-    <ReactTooltip id={labelName.replace(/\s/g, '')+"Tooltip"}>
-      <p>{tooltipText}</p>
-    </ReactTooltip>
-  </div>
+const InfoContainer = styled.div`
+    width: 24px;
+    height: 24px;
+    background-image: url('https://i.ibb.co/1m06xzc/info.png');
```
Comment (rob-hunt-paddle) [inline src/components/info/index.js]: Odd random URL. Worth bringing into the project?

File: src/components/input/index.js [hunk 11]
```diff
@@ -1,53 +1,84 @@
...
-  }
+    render() {
+        const { labelName, tooltipText, updateValue, codeToRender } = this.props
+        return (
+            <InputContainer>
+                <LeftHandSide>
+                    <input
+                        type="checkbox"
+                        onChange={() => this.toggleCheckedParam(labelName)}
+                        checked={this.props.codeToRender.contents.filter(param => param.name === labelName)[0].render}
```
Comment (rob-hunt-paddle) [inline src/components/input/index.js]: Worth using lodash get?

File: src/constants/modal-styles.js [hunk 12]
```diff
@@ -1,25 +1,49 @@
...
-  top: '40px',
-  bottom: '40px',
-  left: '40px',
-  right: '40px',
-  maxWidth: '960px',
-  margin: 'auto',
-  height: '85%'
+    background: '#fff',
+    padding: 0,
+    inset: '0px',
```
Comment (rob-hunt-paddle) [inline src/constants/modal-styles.js]:
```suggestion
    inset: '0',
```

File: src/constants/modal-styles.js [hunk 13]
```diff
@@ -1,25 +1,49 @@
...
+    bottom: 0,
+    left: 0,
+    right: 0,
+    background: 'rgb(0,0,0,0.75)',
+}
+
+export const CONTROL_PANEL_CONTENT = {
+    background: '#fff',
+    padding: 0,
+    inset: '0px',
```
Comment (rob-hunt-paddle) [inline src/constants/modal-styles.js]:
```suggestion
    inset: '0',
```

File: src/utils/greyOutContainer.js [hunk 14]
```diff
@@ -1,3 +1,3 @@
 export const greyContainer = (codeToRender, param) => {
-  return !codeToRender.contents.filter(parameter => parameter.name === param)[0].render
+    return !codeToRender.contents.filter(parameter => parameter.name === param)[0].render
```
Comment (rob-hunt-paddle) [inline src/utils/greyOutContainer.js]: Worth using lodash get?

File: src/constants/dropdown-options.js [hunk 15]
```diff
@@ -1,11 +1,16 @@
 export const LOCALE_OPTIONS = [
-  {name: 'ru', value: 'russian'},
-  {name: 'en', value: 'english'},
-  {name: 'fr', value: 'french'},
-  {name: 'de', value: 'german'},
+    { name: 'russian', value: 'ru' },
```
Comment (rob-hunt-paddle) [inline src/constants/dropdown-options.js]: We're missing a lot of options in this file. Are we planning on filling these out in another PR?

Comment (RichMatthews) [reply src/constants/dropdown-options.js]: yes was planning on opening a PR to do this separately 👍

File: src/utils/greyOutContainer.js (hunk 14 above)
Comment (RichMatthews) [reply src/utils/greyOutContainer.js]: will look into it 👍

File: src/constants/modal-styles.js (hunk 13 above)
Comment (RichMatthews) [reply src/constants/modal-styles.js]: changed 👍

File: src/constants/modal-styles.js (hunk 12 above)
Comment (RichMatthews) [reply src/constants/modal-styles.js]: changed 👍

File: src/components/dropdown/index.js (hunk 4 above)
Comment (RichMatthews) [reply src/components/dropdown/index.js]: will look into it 👍

File: src/components/input/index.js (hunk 11 above)
Comment (RichMatthews) [reply src/components/input/index.js]: will look into it 👍

File: src/components/dropdown/index.js (hunk 3 above)
Comment (RichMatthews) [reply src/components/dropdown/index.js]: changed 👍

File: src/components/Modal/buttons/index.js (hunk 2 above)
Comment (RichMatthews) [reply src/components/Modal/buttons/index.js]: to centralise the text 🙈

File: package.json [hunk 16]
```diff
@@ -1,10 +1,11 @@
 {
   "name": "hackathon",
```
[the remark this answers is not in the document]
Comment (RichMatthews) [reply package.json]: yes 👍

File: src/components/Modal/buttons/index.js (hunk 2 above)
Comment (rob-hunt-paddle) [reply src/components/Modal/buttons/index.js]: That's a hacky way of doing it 😄. Why not use flexbox?

File: src/constants/dropdown-options.js (hunk 15 above)
Comment (rob-hunt-paddle) [reply src/constants/dropdown-options.js]: 👍

File: src/components/Modal/buttons/index.js (hunk 2 above)
Comment (RichMatthews) [reply src/components/Modal/buttons/index.js]: apparently it's the _way_ to do it for buttons? 🤔 🙈

File: src/constants/dropdown-options.js [hunk 17]
```diff
@@ -1,11 +1,16 @@
...
 
-export const CHECKOUT_VERISON_OPTIONS = [
-  {name: 'old', value: 'old'},
-  {name: 'new', value: 'new'},
+export const CHECKOUT_VERISON_OPTIONS = [{ name: 'old', value: 'old' }, { name: 'new', value: 'new' }]
+
+export const COUNTRY_OPTIONS = [{ name: 'england', value: 'england' }]
+
+export const ENV_OPTIONS = [
+    { name: 'dev', value: 'dev' },
```
Comment (rob-hunt-paddle) [inline src/constants/dropdown-options.js]: Should this value be `dev` or `development`?

Comment (RichMatthews) [reply src/constants/dropdown-options.js]: `dev`. well, it works with `dev` 🤷‍♂️

Comment (rob-hunt-paddle) [reply src/constants/dropdown-options.js]: 👍 you're right, paddle.js uses `dev`.

File: src/components/Main/index.js [hunk 18]
```diff
@@ -1,65 +1,82 @@
...
 
-  openCheckout() {
-    window.Paddle.Environment.set("staging");
-    window.Paddle.Setup({ vendor: 33079 });
-    this.refresh()
-  }
+    openCheckout() {
+        window.Paddle.Environment.set(this.props.environment)
+        window.Paddle.Setup({
+            vendor: this.props.codeToRender.contents.filter(item => item.name === 'vendor')[0].value,
```
[the remark this answers is not in the document]
Comment (RichMatthews) [reply src/components/Main/index.js]: changed 👍

File: src/components/dropdown/index.js (hunk 4 above)
Comment (RichMatthews) [reply src/components/dropdown/index.js]: done 👍

File: README.md [hunk 19]
```diff
@@ -1,44 +1,28 @@
...
-### `npm test`
+```
+git clone https://github.com/PaddleHQ/Checkout-Testing-Tool.git
 
-Launches the test runner in the interactive watch mode.<br>
-See the section about [running tests](https://facebook.github.io/create-react-app/docs/running-tests) for more information.
+cd Checkout-Testing-Tool
 
-### `npm run build`
+npm i && npm start
```
Comment (victor-paddle) [inline README.md]: are we using npm or yarn?

Comment (RichMatthews) [reply README.md]: `npm` for now. but we can replace it with yarn in the future 👍

**Pull request merged**